src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -94,73 +117,247 @@ boot_newcomp_sinv_alloc(spdid_t spdid)
 {
-	compcap_t   cc;
-	captblcap_t ct = new_comp_cap_info[spdid].compinfo->captbl_cap;
-	pgtblcap_t  pt = new_comp_cap_info[spdid].compinfo->pgtbl_cap;
-	sinvcap_t   sinv;
-	thdcap_t    main_thd;
-	int         i = 0;
-	unsigned long token = (unsigned long) spdid;
-
-	cc = cos_comp_alloc(&boot_info, ct, pt, (vaddr_t)new_comp_cap_info[spdid].upcall_entry);
+	struct cos_defcompinfo *defci     = cos_defcompinfo_curr_get();
+	struct cos_aep_info    *sched_aep = cos_sched_aep_get(defci);
+	struct cos_aep_info    *child_aep = cos_sched_aep_get(new_comp_cap_info[spdid].defcompinfo);
+	struct cos_compinfo    *child_ci  = cos_compinfo_get(new_comp_cap_info[spdid].defcompinfo);
+
+	child_ci->comp_cap = cc;
+
+	if (comp_flags) {
+		child_aep->thd = cos_initthd_alloc(boot_info(), child_ci->comp_cap);
+		assert(child_aep->thd);
+
+		init_thds[spdid] = child_aep->thd;
+
+		child_aep->tc = cos_tcap_alloc(boot_info());
+		assert(child_aep->tc);
+
+		child_aep->rcv = cos_arcv_alloc(boot_info(), child_aep->thd, child_aep->tc,
+				boot_info()->comp_cap, sched_aep->rcv);
+		assert(child_aep->rcv);
+	} else {
+		child_aep->tc  = sched_aep->tc;
+		child_aep->rcv = sched_aep->rcv;
+	}
+
+	if (comp_flags & BOOT_COMP_FLAG_SCHED) {
+		int ret;
+		/* Give the child infinite tcap if it's the system scheduler */
+		ret = cos_tcap_transfer(child_aep->rcv, sched_aep->tc, TCAP_RES_INF, TCAP_PRIO_MAX+1);
+		if (ret) {
+			printc("tcap transfer to sched init thread failed: %d\n", ret);
+			assert(0);
+		}
+	}
+
+	child_aep->fn   = NULL;
+	child_aep->data = NULL;
+}
+
+static void
+boot_newschedcomp_cap_init(spdid_t spdid, captblcap_t ct, pgtblcap_t pt, compcap_t cc) 
+{
+	struct cos_aep_info *   child_aep = cos_sched_aep_get(new_comp_cap_info[spdid].defcompinfo);
+	pgtblcap_t              untype_pt;
+
+	untype_pt = cos_pgtbl_alloc(boot_info());
+	assert(untype_pt);
+	cos_meminfo_init(&(new_comp_cap_info[spdid].compinfo->mi), BOOT_MEM_KM_BASE,
+			CMP_UTYPMEM_SZ, untype_pt);
+
+	cos_cap_cpy_at(new_comp_cap_info[spdid].compinfo, BOOT_CAPTBL_SELF_INITHW_BASE,
+			boot_info(), BOOT_CAPTBL_SELF_INITHW_BASE);
+	cos_cap_cpy_at(new_comp_cap_info[spdid].compinfo, BOOT_CAPTBL_SELF_CT,
+			boot_info(), ct);
+	cos_cap_cpy_at(new_comp_cap_info[spdid].compinfo, BOOT_CAPTBL_SELF_UNTYPED_PT,
+			boot_info(), untype_pt);
+
+	cos_cap_cpy_at(new_comp_cap_info[spdid].compinfo, BOOT_CAPTBL_SELF_PT, boot_info(), pt);
+	cos_cap_cpy_at(new_comp_cap_info[spdid].compinfo, BOOT_CAPTBL_SELF_COMP, boot_info(), cc);
+
+	/* Populate untyped memory */
+	cos_meminfo_alloc(new_comp_cap_info[spdid].compinfo, BOOT_MEM_KM_BASE, CMP_UTYPMEM_SZ);
+
+	cos_cap_cpy_at(new_comp_cap_info[spdid].compinfo, BOOT_CAPTBL_SELF_INITTHD_BASE,
+			boot_info(), child_aep->thd);
+	cos_cap_cpy_at(new_comp_cap_info[spdid].compinfo, BOOT_CAPTBL_SELF_INITTCAP_BASE,
+			boot_info(), child_aep->tc);
+	cos_cap_cpy_at(new_comp_cap_info[spdid].compinfo, BOOT_CAPTBL_SELF_INITRCV_BASE,
+			boot_info(), child_aep->rcv);
+}
```

Comment:
This whole copying thing is more tricky and I've the updates in my PR anyway. 
Just FYI. Like not every component should be given CT/PT/COMP/UNTYPED_PT etc.